WebUI/lims/AppDatabase.py

```diff
@@ -58,22 +80,72 @@ def executeSql(self, query, connection):
     # Get the replicates based upon if they are running (or not) with an upper limit for how many (default 1000)
     @staticmethod
     def getReplicates(request, running, limit = 1000):
-        SQL = """
-            SELECT id, filename, starttime, endtime, movement,
-                CASE WHEN endtime IS NULL THEN (now() - starttime) ELSE runningtime END AS runningtime
-            FROM v_replicates {} ORDER BY starttime DESC LIMIT %(limit)s"""
+        if running:
+            SQL = """
+                SELECT filename, starttime, movement,
+                    CASE WHEN endtime IS NULL THEN (now() - starttime) ELSE runningtime END AS runningtime
+                FROM v_replicates {} ORDER BY starttime DESC LIMIT %(limit)s"""
+        else:
+            SQL = """
+                SELECT id, filename, starttime, endtime, movement,
+                    CASE WHEN endtime IS NULL THEN (now() - starttime) ELSE runningtime END AS runningtime
+                FROM v_replicates {} ORDER BY starttime DESC LIMIT %(limit)s"""
         WHERE = "WHERE (now() - starttime) <= interval '3 days' AND endtime IS NULL"
 
         # Update the query and return the results
         append = WHERE if running else ""
         return selectQuery(request, SQL.format(append), {'limit': limit})
     
 
+    # Get configurations that associate with specific study id.
+    @staticmethod
+    def getStudyConfigurations(request,studyid = False):
+        # If studyid is a number.
+        if studyid:
+            SQL = """
+                SELECT configuration.filename, 
+                concat('(', ncols, ', ', nrows, ', ', xllcorner, ', ', yllcorner, ', ', cellsize, ')') as spatial, count(replicate.id), configuration.id
+                FROM configuration left join replicate on replicate.configurationid = configuration.id 
+                WHERE studyid = %(id)s group by configuration.id order by configuration.id"""
+            return selectQuery(request, SQL, {'id':studyid})
+        # If studyid is False.
+        else:
+            SQL="""
+                SELECT configuration.filename, 
+                concat('(', ncols, ', ', nrows, ', ', xllcorner, ', ', yllcorner, ', ', cellsize, ')') as spatial, count(replicate.id), configuration.id 
+                FROM configuration left join replicate on replicate.configurationid = configuration.id WHERE studyid is NULL 
+                group by configuration.id order by configuration.id"""
```

Comment:
This is a bit inefficient - since most of the query is unchanged, you can just append the WHERE clause if studyid is not None. There is an example of me doing this elsewhere in the code.